src/js/components/Animate.js

```diff
@@ -0,0 +1,235 @@
+    }, delay);
+  }
+
+  render () {
+    return this.props.children;
+  }
+};
+
+AnimateChild.propTypes = {
+  enter: PropTypes.shape({
+    animation: PropTypes.string,
+    duration: PropTypes.number,
+    delay: PropTypes.number
+  }),
+  leave: PropTypes.shape({
+    animation: PropTypes.string,
+    duration: PropTypes.number,
+    delay: PropTypes.number
+  })
+};
+
+AnimateChild.defaultProps = {
+  enter: {},
+  leave: {}
+};
+
+
+class Animate extends Component {
+  constructor (props) {
+    super(props);
+    this.state = {
+      animationState: 'enter',
+      animation: props.enter.animation
+    };
+  }
+
+  componentWillReceiveProps (nextProps) {
+    const { visible, keep } = this.props;
+    const { animationState } = this.state;
+
+    if (keep && visible !== nextProps.visible) {
+      let state = '';
+      if (!nextProps.visible) {
+        state = (animationState === 'leave') ? 'enter' : 'leave';
+      }
+
+      const animateState = nextProps[state] || nextProps.enter;
+      this.setState({
+        animationState: state,
+        animation: state ? animateState.animation : ''
+      });
+
+      // Reset animation state back to enter after leave animation is finished
+      if (state === 'leave') {
+        const node = ReactDOM.findDOMNode(this);
+        clearTimeout(this.animationTimer);
+        this.animationTimer = setTimeout(() => {
+          this.setState({
+            animationState: 'enter',
+            animation: nextProps.enter.animation
+          });
+        }, (parseFloat(getComputedStyle(node).transitionDuration) +
+        parseFloat(getComputedStyle(node).transitionDelay)) * 1000);
+      }
+    }
+  }
+
+  render () {
+    const {
+      enter,
+      leave,
+      className,
+      children,
+      component,
+      visible,
+      keep,
+      style,
+      ...props
+    } = this.props;
+
+    const animateChildren = React.Children.map(children, (child, index) => {
+      const key = (child && child.key) ? child.key : `animate-${index}`;
+      return (
+        <AnimateChild
+          key={key}
+          enter={enter}
+          leave={leave}
+        >
+          {child}
+        </AnimateChild>
+      );
+    });
+
+    let classes = className;
+    let styles = {...style};
+    if (keep) {
+      classes = classnames(
+        CLASS_ROOT,
+        className,
+        {
+          [`${this.state.animation}--${this.state.animationState}`]: !visible
+        }
+      );
+      styles = {
+        ...style,
+        transitionDuration: `${enter.duration}ms`,
+        transitionDelay: `${enter.delay}ms`
+      };
+    }
+
+    return (
+      <TransitionGroup
+        {...props}
+        className={classes}
+        component={component || 'div'}
+        style={styles}
+      >
+        {(visible || visible === undefined || keep) &&
+          animateChildren
+        }
+      </TransitionGroup>
+    );
+  }
+};
+
+Animate.propTypes = {
+  component: PropTypes.oneOfType([
```

Comment:
I just tried changing this to `PropTypes.node` & I'm receiving a propType warning since i'm passing down Tiles as `component={Tiles}` rather than a React element. I've also tried passing the component as a React element but `ReactTransitionGroup` doesn't like that.